Sources/StreamChat/WebSocketClient/EventMiddlewares/EventConverterMiddleware_Tests.swift

```diff
@@ -0,0 +1,71 @@
+//
+// Copyright © 2021 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+@testable import StreamChat
+@testable import StreamChatTestTools
+import XCTest
+
+final class EventConverterMiddleware_Tests: XCTestCase {
+    var middleware: EventConverterMiddleware!
+    var database: DatabaseContainerMock!
+
+    override func setUp() {
+        middleware = .init()
+        database = try! DatabaseContainerMock(kind: .inMemory)
+        super.setUp()
+    }
+    
+    override func tearDown() {
+        middleware = nil
+        database = nil
+        super.tearDown()
+    }
+    
+    func handle_whenEventDTOComes_toDomainResultIsReturned() throws {
+        class EventDTOMock: EventDTO {
+            let payload = EventPayload(eventType: .channelDeleted)
+            
+            var toDomainEvent_session: DatabaseSession?
+            var toDomainEvent_returnValue: Event?
+            
+            func toDomainEvent(session: DatabaseSession) -> Event? {
+                toDomainEvent_session = session
+                return toDomainEvent_returnValue
+            }
+        }
+        
+        // Create mock event DTO
+        let eventDTO = EventDTOMock()
+        
+        // Hook up the value returned from `toDomainEvent`
+        eventDTO.toDomainEvent_returnValue = EventDTOMock()
+        
+        // Feed event DTO to middleware
+        let result = middleware.handle(event: eventDTO, session: database.viewContext)
+        
+        // Assert the session is forwarded to `toDomainEvent` func
+        XCTAssertEqual(eventDTO.toDomainEvent_session as! NSManagedObjectContext, database.viewContext)
+        
+        // Assert
+        XCTAssertTrue(result as! EventDTOMock === eventDTO.toDomainEvent_returnValue as! EventDTOMock)
+    }
+    
+    func handle_whenNotEventDTOComes_eventIsForwardedAsIs() throws {
+        // Create event
+        let event = UnknownEvent(
+            type: .reactionNew,
+            cid: .unique,
+            userId: .unique,
+            createdAt: .unique,
+            payload: [:]
+        )
+        
+        // Feed event to middleware
+        let result = middleware.handle(event: event, session: database.viewContext)
+        
+        // Assert
+        XCTAssertEqual(result as! UnknownEvent, event)
+    }
```

Comment:
Are these tests even running? 🤔 The func shouldn't start with `test `prefix?

Reply:
🤦

Reply:
good catch, let me fix this 💣

Reply:
✅ Fixed